index.js

```diff
@@ -69,14 +68,16 @@ function getJiraIssueKeys(commitMessages) {
     const retVal = [];
     for (let i = 0; i < commitMessages.length; i++) {
         let commitMessage = commitMessages[i];
-        if (defaultRegex.test(commitMessage)) {
-            const message = defaultRegex.exec(commitMessage)[0];
-            if (retVal.indexOf(message)<0) {
-                retVal[retVal.length] = message;
-            }
+        const re = new RegExp(defaultPattern, "g"); // re-create due to stateful use
+        while ((match = re.exec(commitMessage)) !== null) {

```

Comment:
looks like an infinite loop for me, as in - don't we need to remove match[0] from commitMessage, when we run the next exec?